finetuner/client/base.py

```diff
@@ -1,31 +1,53 @@
+import json
 import os
 from typing import Optional
 
+import hubble
 import requests
 from path import Path
 
-from finetuner.constants import HOST
+from finetuner.constants import (
+    AUTHORIZATION,
+    CHARSET,
+    DATA,
+    HOST,
+    HUBBLE_USER_ID,
+    TOKEN_PREFIX,
+    UTF_8,
+)
 
 
 class BaseClient(object):
-    """Base Finetuner API client.
+    """Base Finetuner API client."""
 
-    :param user_id: Unique identifier of a user.
-    """
-
-    def __init__(self, user_id):
-        self._user_id = user_id
-        self._session = requests.session()
+    def __init__(self):
         self._base_url = Path(os.environ.get(HOST))
-        self._session.headers.update({'Accept-Charset': 'utf-8'})
+        self._session = self._get_client_session()
+        self._hubble_client = hubble.Client(max_retries=None, timeout=10, jsonify=True)
+        self._hubble_user_id = self._get_hubble_user_id()
+
+    def _get_hubble_user_id(self):
+        user_info = json.loads(self._hubble_client.get_user_info())
+        if user_info['code'] >= 400:
+            # will implement error-handling later
+            pass
+        hubble_user_id = user_info[DATA][HUBBLE_USER_ID]
+        return hubble_user_id
+
+    @staticmethod
+    def _get_client_session() -> requests.Session:
+        session = requests.Session()
+        api_token = TOKEN_PREFIX + str(hubble.Auth.get_auth_token())
+        session.headers.update({CHARSET: UTF_8, AUTHORIZATION: api_token})
+        return session
 
     def handle_request(
         self,
         url: str,
         method: str,
         params: Optional[dict] = None,
         json: Optional[dict] = None,
-    ):
+    ) -> requests.Response:
```

Comment:
I think we should return dicts. No reason for the user to have to deal with custom `requests` objects

Reply:
Agree, they should at least have option to receive dicts. But the PR is too big now and maybe we should address it next time?